### PR TITLE
Combine Product Detail thumbnail and main image into one carousel/image

### PR DESCRIPTION
**Pull request by starkiller4996**
### Issues Fixed
https://github.com/JumpMind/openpos-framework/issues/2562

## Desktop (1366x768):
carousel-register

singleImage-register

## Tablet (1180x820):
singleImage-tabletL

singleImage-tabletP

carousel-tabletL

carousel-tabletP

## Mobile (Zebra - 360x640):
singleImage-mobileL

singleImage-mobileP

carousel-mobileL

carousel-mobileP

### CODE REVIEW

**Comment by jrmihalick** [comment]
@starkiller4996 FYI, Petco Zebra res is 360x720

**Review by jrmihalick — APPROVED** [review]
Looks good to me.  Hopefully the user can scroll down on the mobile screen to see the product attributes (e.g., Promotions)?

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/screens-with-parts/product/product-detail/product-detail.component.html [hunk 1]
```diff
@@ -1,9 +1,15 @@
 <app-bacon-strip>
     <div class="content-container" responsive-class>
         <div class="product-header" responsive-class>
-            <app-image-carousel class="carousel" [imgUrls]="screen.imageUrls" [altImageUrl]="screen.alternateImageUrl"
-                [verticalThumbprints]="false" altImageText="Image Not Found" responsive-class>
-            </app-image-carousel>
+            <app-carousel [itemsPerSlide]="1" [navigationArrowsSize]="'sm'">
+                <ng-template *ngFor="let category of screen.imageUrls; let i = index" #carouselItem>
+                    <div class="carousel-image" responsive-class>
+                        <app-image [imageUrl]="screen.imageUrls[i]"
```
Comment (jchirico8) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/screens-with-parts/product/product-detail/product-detail.component.html:7]: Isn't screen.imageUrls[i] the same as category?

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/carousel/carousel.component.scss [hunk 2]
```diff
@@ -14,6 +14,7 @@
     overflow: hidden;
     // This can be overridden with a div.item-container selector in the consumers theme file
     height: 9rem;
+    width: 300px;
```
Comment (jchirico8) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/carousel/carousel.component.scss:17]: I don't like a hard width here. This component is used for multiple things that shouldn't have to be fixed later to overwrite the width

**Review by jchirico8 — CHANGES_REQUESTED** [review]

**Review by jrmihalick — APPROVED** [review]

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/carousel/carousel.component.ts [hunk 3]
```diff
@@ -22,6 +22,7 @@ export class CarouselComponent implements AfterContentInit, OnChanges, OnDestroy
     @Input() wrapNavigationToFront = true;
     @Input() wrapNavigationToBack = true;
     @Input() itemsPerSlide = 1;
+    @Input() width = 300;
```
Comment (jchirico8) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/carousel/carousel.component.ts:25]: This is still defaulting to 300 and requiring any other consumer to override it. The 300px width that product detail wants is the exception and should be the only place setting it

**Review by jchirico8 — APPROVED** [review]

**Pull request merged**